eq_cir_management_ui/errors/routes.py

```diff
@@ -0,0 +1,63 @@
+"""Errors routes."""
+
+from flask import Blueprint, render_template, request
+from structlog import get_logger
+from werkzeug.exceptions import (
+    Forbidden,
+    InternalServerError,
+    NotFound,
+    Unauthorized,
+)
+
+logger = get_logger()
+
+errors_blueprint = Blueprint("errors", __name__)
+
+
+def log_exception(exception: Exception, status_code: int) -> None:
+    """Log the exception with the appropriate log level based on the status code."""
+    log = logger.warning if status_code < 500 else logger.error
+
+    log(
+        "an error has occurred",
+        exc_info=exception,
+        url=request.url,
+        status_code=status_code,
+    )
+
+
+@errors_blueprint.app_errorhandler(401)
+def unauthorized(exception: Unauthorized) -> tuple[str, int]:
+    """401 page.
+    :return: Rendered HTML.
+    """
+    log_exception(exception, 401)
+
+    return render_template("errors/401.html"), 401
```

Comment:
Not sure if we are not overcomplicating this here if we really only need same template, one template - `error.html` with different `title` and `paragraphs`/`contents` vars being passed in. There is no structural difference between these error pages, only contents. Something to discuss with the devs possibly?

Reply:
I've refactored this now, probably how I should have done it in the first place, to create a 'base-error' template, that extends the top level 'base' template, and contains the breadcrumb etc, so that now the individual templates only contain the specific content for that error.
I've also added a configuration that allows for specific page titles to be set on the routes, with a fallback default when one is not provided.

Reply:
I was thinking we could make it even simpler, so in template something like:
```jinja
<h1 class="ons-u-mb-xl">{{ heading -}}</h1>
  <p class="ons-u-mb-xl">{{ paragraph -}}</p>
```
or even and pass in the paragraph's text list/array:
```jinja
<h1 class="ons-u-mb-xl">{{ heading -}}</h1>
{%- for paragraph in paragraphs %}
    <p class="ons-u-mb-xl">{{ paragraph -}}</p>
{% endfor %}
  
```
Then you end up with one error template. As long as are error pages are simple, we don't need this extra repetition.

Reply:
I think that this is abstracting the content too far, and over complicating the implementation.
The suggested approach works well when the content is variable and coming from db/json, or requires translation, but in this case the content is static, so defining the page text in simple html files seems like the simplest solution.
I can also see the potential need to extend the 500 template to include the error message, adding a Details or Panel component, which just adds to the complexity.
The content has to be defined somewhere regardless of approach. I think the individual file approach seems like the most straight forward and flexible option.

Reply:
Following Tech Session discussion, have refactored to introduce single error template